sources/ClangSharpPInvokeGenerator/Program.cs

```diff
@@ -777,23 +777,9 @@ public static void Run(InvocationContext context)
                     context.Console.Write("    ");
                     context.Console.WriteLine(diagnostic.ToString());
 
-                    if (diagnostic.Level == DiagnosticLevel.Warning)
+                    if (diagnostic.Level == DiagnosticLevel.Error)
```

Comment:
This prior behavior is intentional because it did not exit with "success".

There is:
* succeeded
* exited with warnings
* exited with errors

Where error means something catastrophic happened and you should not trust the results and where warning means some kind of semi-recoverable error happened and the results are likely incorrect in more than one way.

While there is not currently a way to suppress individual warnings directly, there are options to suppress particular categories of warnings. For example you can avoid various diagnostics by using the "exclusion options": https://github.com/dotnet/ClangSharp?tab=readme-ov-file#generating-bindings. This includes both `-c ...` (for broad configuration) and `-e ...` (for explicit declaration/statement/etc exclusion).

If there are other particular classes of warnings that should be considered for exclusion, the "right" fix would be to provide new switches allowing those broad exclusions (whether it be via particular diagnostic IDs or via things that indicate that function-like macros should be ignored, for example).